### PR TITLE
Add initial code for Google Ads connector

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Add initial code for Google Ads connector

### CODE REVIEW

File: src/main/resources/functions/adGroupAdsMutate.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <sequence>
+        <class name="org.wso2.carbon.google.ads.connector.RestURLBuilder">
+            <property name="operationPath" value="/v17/customers/{customerId}/adGroupAds:mutate"/>
+            <property name="pathParameters" value="customerId,"/>
+        </class>
+        <payloadFactory media-type="json" template-type="freemarker">
+            <format>
+                <![CDATA[{
+                                                                                        <#if (args.arg1)?has_content>
+                                    "operations": ${args.arg1},                                </#if>
```
Comment (pcnfernando) [inline src/main/resources/functions/adGroupAdsMutate.xml:43]: Can use following to mitigate optional args causing an invalid json. If the json being created has to be a nested one, use typeching and then dump the map. Update the gen tool accordingly

```ftl
<#assign argMap = {}>

<#if (args.arg1)?has_content>
  <#assign argMap = argMap + {"country": args.arg1}>
</#if>

<#if (args.arg2)?has_content>
  <#assign argMap = argMap + {"city": args.arg2}>
</#if>

<#if (args.arg3)?has_content>
  <#assign argMap = argMap + {"city": args.arg3}>
</#if>


{
<#list argMap as k,v>
 "${k}" : "${v}"<#sep>,
</#list>
}
```

File: src/main/java/org/wso2/carbon/google/ads/connector/ClientCredentialsAccessTokenHandler.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                TokenManager.removeToken(getTokenKey(connectionName, tokenEndpoint, developerToken, payloadParametersMap));
+            }
+            if (log.isDebugEnabled()) {
+                if (token == null) {
+                    log.debug("Token does not exists in token store.");
+                } else {
+                    log.debug("Access token is inactive.");
+                }
+            }
+            token = getAndAddNewToken(messageContext, connectionName, developerToken, payloadParametersMap, tokenEndpoint);
```
Comment (pcnfernando) [inline src/main/java/org/wso2/carbon/google/ads/connector/ClientCredentialsAccessTokenHandler.java:97]: Move `getTokenKey()` to var and reuse to get and remove.

File: src/main/java/org/wso2/carbon/google/ads/connector/ClientCredentialsAccessTokenHandler.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            Utils.setErrorPropertiesToMessage(messageContext, Constants.ErrorCodes.INVALID_CONFIG, ERROR_MESSAGE);
+            handleException(ERROR_MESSAGE, messageContext);
+        }
+
+        Map<String, String> payloadParametersMap = new HashMap<>();
+        payloadParametersMap.put(Constants.OAuth2.REFRESH_TOKEN, refreshToken);
+        payloadParametersMap.put(Constants.OAuth2.CLIENT_ID, clientId);
+        payloadParametersMap.put(Constants.OAuth2.CLIENT_SECRET, clientSecret);
+
+        Token token = TokenManager.getToken(getTokenKey(connectionName, tokenEndpoint, developerToken, payloadParametersMap));
```
Comment (pcnfernando) [inline src/main/java/org/wso2/carbon/google/ads/connector/ClientCredentialsAccessTokenHandler.java:85]: Move `getTokenKey()` to var and reuse to get and remove.

Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/google/ads/connector/ClientCredentialsAccessTokenHandler.java]: Fixed with [fa7b308](https://github.com/wso2-extensions/esb-connector-googleads/pull/1/commits/fa7b308f94a8a150e1d96097a6df04f4378772c0)

File: src/main/resources/functions/adGroupAdsMutate.xml (hunk 1 above)
Comment (SanojPunchihewa) [reply src/main/resources/functions/adGroupAdsMutate.xml:43]: @arunans23 Please incorporate this improvement when designing the new gen tool.

File: src/main/resources/config/init.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <parameter name="loginCustomerId" description="The customer ID of the Manager Account."/>
+    <sequence>
+        <class name="org.wso2.carbon.google.ads.connector.ClientCredentialsAccessTokenHandler"/>
+        <header name="developer-token" expression="$func:developerToken" scope="transport"/>
+        <filter source="boolean($func:loginCustomerId)" regex="true">
+            <then>
+                <header name="login-customer-id" expression="$func:loginCustomerId" scope="transport"/>
+            </then>
+        </filter>
+        <header name="Authorization" expression="fn:concat('Bearer ', $ctx:_OH_INTERNAL_ACCESS_TOKEN_)" scope="transport"/>
```
Comment (arunans23) [inline src/main/resources/config/init.xml:37]: Is this something related to Healthcare connectors?

Comment (SanojPunchihewa) [reply src/main/resources/config/init.xml:37]: `_OH_INTERNAL_ACCESS_TOKEN_ ` was the property used in the Java class to set the access token. I used the same one generated by the tool. We can rename it.

Comment (arunans23) [reply src/main/resources/config/init.xml:37]: Since this is internal, we can keep it as it is.

**Review by arunans23 — APPROVED** [review]
